src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsGrid.component.ts

```diff
@@ -49,18 +57,117 @@ export class NotebookViewsGridComponent implements OnInit {
 
 	ngAfterViewInit() {
 		const self = this;
+		this._activeView = this.activeView;
 
-		self._grid = GridStack.init({
+		this.createGrid();
+
+		this._loaded = true;
+		this.detectChanges();
+
+		self._grid.on('added', function (e: Event, items: GridStackNode[]) { if (self._gridEnabled) { self.persist('added', items, self._grid, self._items); } });
+		self._grid.on('removed', function (e: Event, items: GridStackNode[]) { if (self._gridEnabled) { self.persist('removed', items, self._grid, self._items); } });
+		self._grid.on('change', function (e: Event, items: GridStackNode[]) { if (self._gridEnabled) { self.persist('change', items, self._grid, self._items); } });
+	}
+
+	ngAfterContentChecked() {
+		if (!this._activeView || this.activeView.guid !== this._activeView.guid) {
```

Comment:
Some comments here explaining this would be nice. 

Are you purposely not clearing the activeView? And then I'm a little confused about why you have two separate vars for activeView.